### PR TITLE
Update to null safety migration

### PR DESCRIPTION
**Pull request by mit-mit**

### CODE REVIEW

File: src/null-safety/migration-guide.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ The instructions assume your code is under **source control**,
 so that you can easily undo any changes.
 
 
-### Switch to a 2.12 beta release
+### Switch to the latest 2.12 beta release
```
Comment (kevmoo) [inline src/null-safety/migration-guide.md:66]: We need to call out Flutter explicitly, right?

We need to make sure users use *at least* the latest beta, but *NOT* dev or master. Or they risk messing things up!

File: src/null-safety/migration-guide.md [hunk 2]
```diff
@@ -133,26 +133,11 @@ You can find contact details on the package page on [pub.dev][].
...
-     sdk: '>=2.8.1 <3.0.0'
-
-   dependencies:
-     path: ^1.8.0-nullsafety
-     process: ^4.0.0-nullsafety
-
-   dev_dependencies:
-     pedantic: ^1.10.0-nullsafety
-   ```
+1. Run `dart pub upgrade --null-safety` to upgrade to the latest versions
```
Comment (kevmoo) [inline src/null-safety/migration-guide.md:136]: Do we want to tell folks to explicitly set *stable* versions as lower bounds?

**Comment by kevmoo** [comment]
Should we give a heads up to ~ignore this message, too?

Package validation found the following potential issue:
* Packages with an SDK constraint on a pre-release of the Dart SDK should themselves be published as a pre-release version. If this package needs Dart version 2.12.0-0, consider publishing the package as a pre-release instead.
  See https://dart.dev/tools/pub/publishing#publishing-prereleases For more information on pre-releases.

File: src/null-safety/migration-guide.md (hunk 1 above)
Comment (mit-mit) [reply src/null-safety/migration-guide.md:66]: I'm not sure I follow; the text below already 1) says to pick Dart or Flutter, and 2) to use beta ?

File: src/null-safety/migration-guide.md (hunk 2 above)
Comment (mit-mit) [reply src/null-safety/migration-guide.md:136]: Done (under the `### Package version` section)

**Comment by mit-mit** [comment]
> Should we give a heads up to ~ignore this message, too?

Yes, added under the `### Package version` section

**Review by kwalrath — COMMENTED** [review]
I started this and then got distracted by something else, but I'm going to send it anyway, because of our time difference. I'll finish my review later.

File: src/null-safety/migration-guide.md [hunk 3]
```diff
@@ -473,38 +457,29 @@ With these constraints,
...
 
-For subsequent updates to the null-safe prerelease of the package,
-increment the prerelease suffix.
-For example, if the first null-safe version is `3.0.0-nullsafety.0`,
-then the next one is `3.0.0-nullsafety.1`.
+As null safety is considered API stable as of **TODO: beta SDK version**, you
+can publish new stable versions of your package, even though null safety isn't
+yet released in a stable Dart SDK release. When published to pub.dev, the site
+will tag your new null safety version as a "Preview", as illustrated for version
+`2.0.0` of `package:args` here:
```
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/null-safety/migration-guide.md:471]: We could point to the published docs for preview releases.

File: src/null-safety/migration-guide.md [hunk 4]
```diff
@@ -473,38 +457,29 @@ With these constraints,
...
-* If your package hasn't reached 1.0.0 yet,
-  _either_ increase the minor version _or_ update the version to 1.0.0.
-  For example, if the previous version is `0.3.2`,
-  the new version is one of the following:
-  * **`0.4.0-nullsafety.0`**
-  * **`1.0.0-nullsafety.0`**
+Update the version of the package to a new major version to indicate a breaking
+change. If your package is already at 1.0.0 or greater, increase the major
+version, for example from `2.3.2` to `3.0.0`. If your package hasn't reached
+1.0.0 yet, _either_ increase the minor version _or_ update the version to 1.0.0.
```
[the remark this answers is not in the document]
Comment (mit-mit) [reply src/null-safety/migration-guide.md]: Re: "but the last sentence says you could only increase the minor version."

For a package which is not yet at 1.0.0, a minor version increment is considered a semantic bump. So `0.3.2` to `0.4.0`, which is a minor version increment, is a semantic bump, just like `1.3.2` to `2.0.0` is.

Is there a better way we can phrase that?

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/null-safety/migration-guide.md]: "to indicate a breaking change" (in the first sentence of the intro paragraph) isn't sufficient?

**Comment by mit-mit** [comment]
I'll be out the next week. Feel free to edit and land this PR!

File: src/null-safety/migration-guide.md (hunk 1 above)
Comment (kwalrath) [reply src/null-safety/migration-guide.md:66]: Could we just say "Switch to the latest beta release"?

**Comment by kwalrath** [comment]
@kevmoo I made some copy edits... please check my work! Also, we should probably be consistent with https://dart.dev/tools/pub/publishing#publishing-previews... and maybe link to/from it? Maybe remove some content from here if it isn't essential?

**Comment by kwalrath** [comment]
Staged: https://kw-staging-dartlang-2.firebaseapp.com/null-safety/migration-guide

**Comment by kevmoo** [comment]
I confirmed that the SDK version shows – "2.12.0-259.9.beta" – works with the currently deployed pub version. So I think we're good.

We should do a follow-up to remove the `-nullsafety` things in the images, though.

**Review by kevmoo — APPROVED** [review]

**Pull request merged**